server/web_ui/src/views/identityverification.rs

```diff
@@ -0,0 +1,529 @@
+                    }
+                    IdentifyUserState::SubmitCodeFirst { .. } => {
+                        self.state = IdentifyUserState::DisplayCodeSecond {
+                            self_totp: totp,
+                            step,
+                        };
+                    }
+                    _ => self.set_state_to_corrupt_state_err(),
+                }
+            }
+            IdentifyUserTransition::WaitForCode => {
+                // here again we have two possibilities: we either come from IdDisplayAndSubmit or from DisplayCodeFirst
+                // if we are in the first case then we go to SubmitCodeFirst, otherwise we go to SubmitCodeSecond
+                match &self.state {
+                    IdentifyUserState::IdDisplayAndSubmit => {
+                        self.state = IdentifyUserState::SubmitCodeFirst {
+                            other_totp: None,
+                            totp_valid: false,
+                        };
+                    }
+                    IdentifyUserState::DisplayCodeFirst { .. } => {
+                        self.state = IdentifyUserState::SubmitCodeSecond {
+                            other_totp: None,
+                            totp_valid: false,
+                        };
+                    }
+                    _ => self.set_state_to_corrupt_state_err(),
+                }
+            }
+            IdentifyUserTransition::Success => match self.state {
+                IdentifyUserState::DisplayCodeSecond { .. }
+                | IdentifyUserState::SubmitCodeSecond { .. } => {
+                    self.state = IdentifyUserState::Success;
+                }
+                _ => self.set_state_to_corrupt_state_err(),
+            },
+            IdentifyUserTransition::DoNothing => return false,
+            IdentifyUserTransition::CheckInput { input } => {
+                // according to our beautiful state machine if CheckInput was called we must be in either IdDisplayAndSubmit, SubmitCodeFirst or SubmitCodeSecond.
+                // if that's the case then we just update the valid_status accordingly
+                // If we're in another state we'll land in the infamous invalid flow error!
+                match &mut self.state {
+                    IdentifyUserState::IdDisplayAndSubmit => self.other_id = input,
+                    IdentifyUserState::SubmitCodeFirst {
+                        other_totp,
+                        totp_valid,
+                    }
+                    | IdentifyUserState::SubmitCodeSecond {
+                        other_totp,
+                        totp_valid,
+                    } => {
+                        *totp_valid = VALIDATE_TOTP_RE.is_match(&input);
+                        #[cfg(debug_assertions)]
+                        console::debug!(input.clone());
+                        *other_totp = input.parse::<u32>().ok(); //this should never fail as we set the input type to "number"
```

Comment:
Yeah but even if it did when I unwrap the option I use `unwrap_or_default`, here with `.ok` I'm just transforming the Result into Option so I don't think it will panic even if it's `Err`. But yeah prob I should remove the comment then

Reply:
That's what I get for reviewing code at midnight, you're right the `.ok` coerces it into a `None`, my bad 😊 

The comment's misleading however 👍🏻

Reply:
fixed it :)